File: ChatGPT(예진)/Python (수정 완료)/Finding_the_Street_Between_Stores.py
import math

# Define the Store class
class Store:
    def __init__(self):
        self.name = ""
        self.x = 0.0
        self.y = 0.0

# Function to input store locations
def input_store_locations(stores):
    for i in range(3):
        stores[i].name = input("Enter the store name: ")
        stores[i].x = float(input("Enter the x-coordinate of the store: "))
        stores[i].y = float(input("Enter the y-coordinate of the store: "))

# Function to calculate the Euclidean distance between two points
def euclidean_distance(point1, point2):
    return math.sqrt((point1.x - point2.x) ** 2 + (point1.y - point2.y) ** 2)

# Function to find the closest and farthest two stores
def find_closest_and_farthest(stores):
    min_distance = float('inf')
    max_distance = float('-inf')
    closest_stores = ["", ""]
    farthest_stores = ["", ""]

    for i in range(3):
        for j in range(i + 1, 3):
            distance = euclidean_distance(stores[i], stores[j])
            if distance < min_distance:
                min_distance = distance
                closest_stores[0] = stores[i].name
                closest_stores[1] = stores[j].name
            if distance > max_distance:
                max_distance = distance
                farthest_stores[0] = stores[i].name
                farthest_stores[1] = stores[j].name

    return closest_stores, min_distance, farthest_stores, max_distance

# Main function
def main():
    stores = [Store() for _ in range(3)]

    print("Enter the locations of three stores.")
    input_store_locations(stores)
    closest_stores, min_distance, farthest_stores, max_distance = find_closest_and_farthest(stores)
    print(f"The closest two stores are: {closest_stores[0]}, {closest_stores[1]} Distance: {min_distance:.2f}")
    print(f"The farthest two stores are: {farthest_stores[0]}, {farthest_stores[1]} Distance: {max_distance:.2f}")

if __name__ == "__main__":
    main()

File: ChatGPT(예진)/C (수정 완료)/Finding_the_Street_Between_Stores.c
#define _CRT_SECURE_NO_WARNINGS
#include <stdio.h>
#include <math.h>

// Define the Store structure
struct Store {
    char name[50];
    float x;
    float y;
};

// Function to input store locations
void input_store_locations(struct Store stores[]) {
    for (int i = 0; i < 3; i++) {
        printf("Enter the store name: ");
        scanf("%s", stores[i].name);
        printf("Enter the x-coordinate of the store: ");
        scanf("%f", &stores[i].x);
        printf("Enter the y-coordinate of the store: ");
        scanf("%f", &stores[i].y);
    }
}

// Function to calculate the Euclidean distance between two points
float euclidean_distance(struct Store point1, struct Store point2) {
    return sqrt(pow((point1.x - point2.x), 2) + pow((point1.y - point2.y), 2));
}

// Function to find the closest and farthest two stores
void find_closest_and_farthest(struct Store stores[], char closest_stores[][50], float* min_distance, char farthest_stores[][50], float* max_distance) {
    *min_distance = INFINITY;
    *max_distance = -INFINITY;

    for (int i = 0; i < 3; i++) {
        for (int j = i + 1; j < 3; j++) {
            float distance = euclidean_distance(stores[i], stores[j]);
            if (distance < *min_distance) {
                *min_distance = distance;
                sprintf(closest_stores[0], "%s", stores[i].name);
                sprintf(closest_stores[1], "%s", stores[j].name);
            }
            if (distance > *max_distance) {
                *max_distance = distance;
                sprintf(farthest_stores[0], "%s", stores[i].name);
                sprintf(farthest_stores[1], "%s", stores[j].name);
            }
        }
    }
}

// Main function
int main() {
    struct Store stores[3];
    char closest_stores[2][50];
    char farthest_stores[2][50];
    float min_distance, max_distance;

    printf("Enter the locations of three stores.\n");
    input_store_locations(stores);
    find_closest_and_farthest(stores, closest_stores, &min_distance, farthest_stores, &max_distance);
    printf("The closest two stores are: %s, %s Distance: %.2f\n", closest_stores[0], closest_stores[1], min_distance);
    printf("The farthest two stores are: %s, %s Distance: %.2f\n", farthest_stores[0], farthest_stores[1], max_distance);

    return 0;
}
